---
title: "Epigenetic and MZT Biomarker Expression"
author: "Erin Chille"
date: "Last updated: 2020/11/09"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Set up workspace

Remove prior objects and load necessary tools
```{r, warning=FALSE, message=FALSE}
rm(list=ls()) # removes all prior objects

library("RColorBrewer")
library("ggplot2")
library("tidyverse")
library("ggpubr")
library("DESeq2")
```


Load sample Information
```{r}
sample.info <- read.csv(file="Sample_Info/RNAseq_data.csv", header=T, sep=",") #load sample info
```

Load gene count Information
```{r}
counts <- as.data.frame(read.csv("1-QC-Align-Assemble/Output/gene_count_matrix.csv", row.names="gene_id"), colClasses = double) #Load normalized expression matrix from stringtie
```


Keep only ambient treatmentinfo and count data
```{r}
dev <- c("AMB")
treatmentinfo_dev <- filter(sample.info, treatment %in% dev)
dim(treatmentinfo_dev) #rows should be 24

gcount_dev <- counts[treatmentinfo_dev$sample_id]
dim(gcount_dev) #columns should be 24
```

### Read normalization
We are now going normalize our read counts using VST-normalization in DESeq2

#### Construct the DESeq2 dataset

Merge the treatment and time_point columns into a new column , group. Set group as a factor.
```{r}
treatmentinfo_dev$time_point <- factor(treatmentinfo_dev$time_point, levels = c("Unfertilized_egg", "Fertilized_egg", "Cleavage", "Prawn_chip", "Early_gastrula", "Mid_gastrula", "Late_gastrula", "Planula", "Adult"))
```

Create a DESeqDataSet design from gene count matrix and labels. Here we set the design to look at time_point to test for any differences in gene expression across timepoints.
```{r}
#Set DESeq2 design
gdds_dev <- DESeqDataSetFromMatrix(countData = gcount_dev,
                              colData = treatmentinfo_dev,
                              design = ~time_point)
```

#### Log-transform the count data
First we are going to log-transform the data using a variance stabilizing transforamtion (VST). This is only for visualization purposes. Essentially, this is roughly similar to putting the data on the log2 scale. It will deal with the sampling variability of low counts by calculating within-group variability (if blind=FALSE). Importantly, it does not use the design to remove variation in the data, and so can be used to examine if there may be any variability do to technical factors such as extraction batch effects.

To do this we first need to calculate the size factors of our samples. This is a rough estimate of how many reads each sample contains compared to the others. In order to use VST (the faster log2 transforming process) to log-transform our data, the size factors need to be less than 4. Otherwise, there could be artefacts in our results.
```{r}
SF.gdds_dev <- estimateSizeFactors(gdds_dev) #estimate size factors to determine if we can use vst  to transform our data. Size factors should be less than for to use vst
print(sizeFactors(SF.gdds_dev)) #View size factors
```

Our size factors are all less than 4, so we can use VST!
```{r}
gvst_dev <- vst(gdds_dev, blind=FALSE) #apply a variance stabilizing transforamtion to minimize effects of small counts and normalize wrt library size
head(assay(gvst_dev), 3) #view transformed gene count data
```
Find mean, median, mode of expression
```{r}
#Make a list of observations
exp_df <- as.data.frame(assay(gvst_dev))
exp_list <- c(exp_df$X119, exp_df$X120, exp_df$X121, exp_df$X127, exp_df$X132, exp_df$X153, exp_df$X154, exp_df$X159, exp_df$X162, exp_df$X163, exp_df$X167, exp_df$X179, exp_df$X180, exp_df$X184, exp_df$X212, exp_df$X215, exp_df$X218, exp_df$X221, exp_df$X359, exp_df$X361, exp_df$X375, exp_df$X1101, exp_df$X1548, exp_df$X1628)
sanity.check <- nrow(exp_df)*ncol(exp_df) #make sure num observations match
length(exp_list) == sanity.check #make sure num observations match. Should be true.

#Get stats and view distribution as a histogram
max(exp_list)
mean(exp_list)
median(exp_list)
min(exp_list)
hist(exp_list)
```


## Run BLAST search of biomarkers against Mcap protein fasta

Find orthologues of biomarkers within Mcap genome
```{bash}
#Make blast database from Mcap predicted protein fasta file
#/usr/local/ncbi/blast/bin/makeblastdb -in {path2ref}/Mcap.protein.fa -input_type fasta -dbtype prot -parse_seqids -out 2b-Epi-MZT-Enzyme-Expression/Input/Mcap.protein.blast.db

#Blast fasta file containing sequences of selected biomarkers against our sequences
#/usr/local/ncbi/blast/bin/blastp -query 2b-Epi-MZT-Enzyme-Expression/Input/cnidarian.Epi_MTZ_biomarkers.fa -db 2b-Epi-MZT-Enzyme-Expression/Input/Mcap.protein.blast.db -outfmt 6 -max_target_seqs 100 -evalue 1e-05 -out 2b-Epi-MZT-Enzyme-Expression/Output/Mcap_Epi_MZT_biomarker_hits_100.tsv

#Manually in text editor: (1.) Add column names: "queryprot", "gene_id", "pident", "length", "mismatch", "gapopen", "qstart", "qend", "sstart", "send", "evalue", "bitscore", and (2.) Remove ("augustus." and ".t1") from gene_ids using "command F" replace
#Manually in Excel: Add a column called Epi.gene that lists the alias for each of your target of enzymes. Check the annotation to see if there is any overlap in genes
```

Subset results for genes of interest
```{r}
hits <- read.table(file="2b-Epi-MZT-Enzyme-Expression/Output/cnidarian.Mcap_Epi_MZT_hits_100.csv", header=T, sep=",")
hits <- hits %>% distinct(Epi.gene, gene_id, .keep_all = TRUE)
hits <- hits %>% group_by(Epi.gene) %>% distinct(sstart, ssend, .keep_all = TRUE)
hits <- filter(hits, bitscore>50)

levels(as.factor(hits$queryprot))

countsx <- as.data.frame(assay(gvst_dev))
gene.counts <- countsx[(row.names(countsx) %in% hits$gene_id), ]
#str(gene.counts)
gene.counts$gene_id <- row.names(gene.counts) 
gene.counts <- merge(gene.counts, hits[,c(1,3)], by="gene_id")

gene.counts.long <- pivot_longer(gene.counts, 2:25, values_to = "exp")
  #gather(gene.counts, group, exp, X1101:X375, factor_key=TRUE) 
colnames(gene.counts.long)[3] <- "sample_id"
gene.counts.long <- merge(gene.counts.long, sample.info[,c(1:2)], by="sample_id")
#str(gene.counts.long)

mean.counts <- gene.counts.long %>%
  group_by(gene_id, Epi.gene, time_point) %>%
  summarise(mean = mean(exp),
            sd=sd(exp))

mean.counts$time_point <- factor(mean.counts$time_point, 
                                 levels = c("Unfertilized_egg","Fertilized_egg","Cleavage", 
                                  "Prawn_chip","Early_gastrula","Mid_gastrula", "Late_gastrula",
                                  "Planula", "Adult"))
levels(as.factor(mean.counts$Epi.gene))
```

```{r}
DNMT3.counts <- subset(mean.counts, Epi.gene =="DNMT3A")
nDNMT3 <- length(levels(as.factor(DNMT3.counts$gene_id)))
DNMT1.counts <- subset(mean.counts, Epi.gene =="DNMT1")
nDNMT1 <- length(levels(as.factor(DNMT1.counts$gene_id)))
TET1.counts <- subset(mean.counts, Epi.gene =="TET1")
nTET <- length(levels(as.factor(TET1.counts$gene_id)))
MBD2.counts <- subset(mean.counts, Epi.gene =="MBD2")
nMBD2 <- length(levels(as.factor(MBD2.counts$gene_id)))
MBD3.counts <- subset(mean.counts, Epi.gene =="MBD3")
nMBD3 <- length(levels(as.factor(MBD3.counts$gene_id)))
UHRF1.counts <- subset(mean.counts, Epi.gene =="UHRF1")
nUHRF1 <- length(levels(as.factor(UHRF1.counts$gene_id)))
BRG1.counts <- subset(mean.counts, Epi.gene =="BRG1")
nBRG1 <- length(levels(as.factor(BRG1.counts$gene_id)))

SMG.counts <- subset(mean.counts, Epi.gene =="SMG") #SMAUG
nSMG <- length(levels(as.factor(SMG.counts$gene_id)))
KAISO.counts <- subset(mean.counts, Epi.gene =="KAISO") #Kaiso
nKAISO <- length(levels(as.factor(KAISO.counts$gene_id)))
TBXT.counts <- subset(mean.counts, Epi.gene =="TBXT") #brachyury
nTBXT <- length(levels(as.factor(TBXT.counts$gene_id)))
SOX2.counts <- subset(mean.counts, Epi.gene =="SOX2") #SOX2
nSOX2 <- length(levels(as.factor(SOX2.counts$gene_id)))
CyclinB.counts <- subset(mean.counts, Epi.gene =="CyclinB") #CyclinB
nCyclinB <- length(levels(as.factor(CyclinB.counts$gene_id)))
WNT8.counts <- subset(mean.counts, Epi.gene =="WNT8") #WNT8
nWNT8 <- length(levels(as.factor(WNT8.counts$gene_id)))

HSP70.counts <- subset(mean.counts, Epi.gene =="HSP70") #HSP70
nHSP70 <- length(levels(as.factor(HSP70.counts$gene_id)))
HSP90.counts <- subset(mean.counts, Epi.gene =="HSP90") #HSP90
nHSP90 <- length(levels(as.factor(HSP90.counts$gene_id)))


nEpihomologues <- as.data.frame(cbind(DNMT3A=nDNMT3, DNMT1=nDNMT1, TET1=nTET, MBD2=nMBD2, MBD3=nMBD3, UHRF1=nUHRF1, BRG1=nBRG1))
print(nEpihomologues)
nMZThomologues <- as.data.frame(cbind(SMG=nSMG, KAISO=nKAISO, TBXT=nTBXT, SOX2=nSOX2, CyclinB=nCyclinB, WNT8=nWNT8))
print(nMZThomologues)
nHSPhomologues <- as.data.frame(cbind(HSP70=nHSP70, HSP90=nHSP90))
print(nHSPhomologues)
```

```{r}
top_DNMT1=c("g53952") #highest eval (0) + bitscore
top_DNMT3=c("g25804") #highest eval (0) + bitscore
#top_MBD=c("g53132") #highest eval (0) + bitscore
#top_UHRF1=c("adi2mcaRNA19502_R1", "adi2mcaRNA19502_R0") # evalue very more sig e-100s versus next e-0s,  and bitscore 5x higher than next genes listed
top_BRG1=c("g68733", "g61145") #evalue 0 and bitscore 3x higher than next genes listed, length twice as long as others listed
top_SMG=c("g4639", "g11098") #longest listed
top_CyclinB=c("g71356") #top 1 longest with best evalue and bitscore
top_KAISO=c("g60350") #longest listed
top_TBXT=c("g68947", "g21139", "g18324") #top 3 have 2x lower evalue and ~150 higher bitscore. Also annotated in transcriptome with BLAST
top_SOX2=c("g53225") #top 1, bitscore and evalue most signif by a lot
top_WNT8=c("g33149") #top 1

top_HSP70=c("adi2mcaRNA1829_R0", "adi2mcaRNA1829_R1")
top_HSP90=c("g33244")
```

## Plot methylation-related genes

Plot DNMT3a expression as a function of time and treatment
```{r}
Mcap.DNMT3 <- DNMT3.counts %>% subset(gene_id%in%top_DNMT3) %>% 
  ggplot(aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(), lty="dashed", color = "darkgrey") +
  geom_point(aes(), size = 1.5) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (vst-transformed counts)"))) +
  theme_bw() + #Set the background color
  ylim(8.728599, 14) +
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_blank(), #set x-axis title size
        axis.text.x=element_blank(), #set x-axis label size
        axis.text.y=element_text(size = 6), #set x-axis label size
        axis.title.y=element_text(size = 6), #set x-axis label size
        axis.ticks.x=element_blank(), #No x-label ticks
        legend.text=element_text(size = 6), #Legend text size
        legend.title=element_text(size = 6), #Legend title size
        legend.position=c(0.75, 0.9)) +
  ggtitle("i) DNMT3A") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 9, 
                                  hjust = 0)) +
  annotate("text", x = 3, y = 14, label = "de novo methylation", size = 2.5)
Mcap.DNMT3
```

Plot DNMT1 expression as a function of time and treatment
```{r}
Mcap.DNMT1 <- DNMT1.counts %>% subset(gene_id%in%top_DNMT1) %>% 
  ggplot(aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(), lty="dashed", color = "darkgrey") +
  geom_point(aes(), size = 1.5) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean normalized expression"))) +
  theme_bw() + #Set the background color
  ylim(8.728599, 14) +
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_blank(), #set x-axis title size
        axis.text.x=element_blank(), #set x-axis label size
        axis.text.y=element_text(size = 6), #set x-axis label size
        axis.title.y=element_blank(), #set x-axis label size
        axis.ticks.x=element_blank(), #No x-label ticks
        legend.text=element_text(size = 6), #Legend text size
        legend.title=element_text(size = 6), #Legend title size
        legend.position=c(0.75, 0.9)) +
  ggtitle("ii) DNMT1") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 9, 
                                  hjust = 0)) +
  annotate("text", x = 3.5, y = 14, label = "methylation maintenance", size = 2.5)
Mcap.DNMT1
```

Plot TET1 expression as a function of time and treatment
```{r}
Mcap.TET1 <- ggplot(TET1.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(), lty="dashed", color = "darkgrey") +
  geom_point(aes(), size = 1.5) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (vst-transformed counts)"))) +
  theme_bw() + #Set the background color
  ylim(8.728599, 14) +
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_blank(), #set x-axis title size
        axis.text.x=element_blank(), #set x-axis label size
        axis.text.y=element_text(size = 6), #set x-axis label size
        axis.title.y=element_blank(), #set x-axis label size
        axis.ticks.x=element_blank(), #No x-label ticks
        legend.text=element_text(size = 6), #Legend text size
        legend.title=element_text(size = 6), #Legend title size
        legend.position=c(0.75, 0.9)) +
  ggtitle("iii) TET1") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 9, 
                                  hjust = 0)) +
  annotate("text", x = 3, y = 14, label = "methylation removal", size = 2.5)
Mcap.TET1
```

Plot MBD2 and MBD3 expression as a function of time and treatment
```{r}
Mcap.MBD2 <- MBD2.counts %>% #subset(gene_id%in%top_MBD) %>% 
  ggplot(aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(lty="dashed", color = "darkgrey") +
  geom_point(aes(), size = 1.5) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (vst-transformed counts)"))) +
  theme_bw() + #Set the background color
  ylim(8.728599, 14) +
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_text(size = 6), #set x-axis title size
        axis.text.x=element_text(angle = 90, vjust = 0.5, hjust=1, size = 6), #set x-axis label size
        axis.text.y=element_text(size = 6), #set x-axis label size
        axis.title.y=element_text(size = 6), #set x-axis label size
        axis.ticks.x=element_blank(), #No x-label ticks
        legend.text=element_text(size = 6), #Legend text size
        legend.title=element_text(size = 6), #Legend title size
        legend.position=c(0.75, 0.9)) +
  ggtitle("iv) MBD2/3") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 9, 
                                  hjust = 0)) +
  annotate("text", x = 3.5, y = 14, label = "transcriptional repression", size = 2.5)
Mcap.MBD2
```

Plot UHRF1 expression as a function of time and treatment
```{r}
#UHRF1.counts %>% subset(gene_id%in%top_UHRF1) %>% 
Mcap.UHRF1 <- ggplot(UHRF1.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(lty="dashed", color = "darkgrey") +
  geom_point(aes(), size = 1.5) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (vst-transformed counts)"))) +
  theme_bw() + #Set the background color
  ylim(8.728599, 14.5) +
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_text(size = 6), #set x-axis title size
        axis.text.x=element_text(angle = 90, vjust = 0.5, hjust=1, size = 6), #set x-axis label size
        axis.text.y=element_text(size = 6), #set x-axis label size
        axis.title.y=element_blank(), #set x-axis label size
        axis.ticks.x=element_blank(), #No x-label ticks
        legend.text=element_text(size = 6), #Legend text size
        legend.title=element_text(size = 6), #Legend title size
        legend.position=c(0.75, 0.85)) +
  ggtitle("v) UHRF1") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 9, 
                                  hjust = 0)) +
  annotate("text", x = 3.5, y = 14.5, label = "methylation maintenance", size = 2.5)
Mcap.UHRF1
```

Plot BRG1 expression as a function of time and treatment
```{r}
Mcap.BRG1 <- BRG1.counts %>% subset(gene_id%in%top_BRG1) %>% 
  ggplot(aes(x=time_point, y=mean, group=gene_id)) +
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(lty="dashed", color = "darkgrey") +
  geom_point(aes(), size = 1.5) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (vst-transformed counts)"))) +
  theme_bw() + #Set the background color
  ylim(8.728599, 14) +
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_text(size = 6), #set x-axis title size
        axis.text.x=element_text(angle = 90, vjust = 0.5, hjust=1, size = 6), #set x-axis label size
        axis.text.y=element_text(size = 6), #set x-axis label size
        axis.title.y=element_blank(), #set x-axis label size
        axis.ticks.x=element_blank(), #No x-label ticks
        legend.text=element_text(size = 6), #Legend text size
        legend.title=element_text(size = 6), #Legend title size
        legend.position = c(0.75, 0.9)) +
  #guides(shape=guide_legend(ncol=2)) +
  ggtitle("vi) BRG1") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic',
                                  size = 9,
                                  hjust = 0)) +
  annotate("text", x = 3.25, y = 14, label = "chromatin accessibility", size = 2.5)
Mcap.BRG1
```

Compile plots into single figure and save
```{r}
Fig.7 <- cowplot::plot_grid(Mcap.DNMT3, Mcap.DNMT1, Mcap.TET1, Mcap.MBD2, Mcap.UHRF1, Mcap.BRG1, align = "vh", ncol = 3)
Fig.7
ggsave("2b-Epi-MZT-Enzyme-Expression/Output/Fig7-Mcap-Dev-Epi-Expression.pdf", Fig.7, width=7, height=5)
```

## Plot MZT biomarkers

Plot CyclinB expression as a function of time and treatment
```{r}
Mcap.CyclinB <- CyclinB.counts %>% subset(gene_id%in%top_CyclinB) %>% 
  ggplot(aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(lty="dashed", color = "darkgrey") +
  geom_point(aes(), size = 1.5) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (vst-transformed counts)"))) +
  theme_bw() + #Set the background color
  #ylim(8,19) +
    theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_blank(), #set x-axis title size
        axis.text.x=element_blank(), #set x-axis label size
        axis.text.y=element_text(size = 6), #set x-axis label size
        axis.title.y=element_text(size = 6), #set x-axis label size
        axis.ticks.x=element_blank(), #No x-label ticks
        legend.title=element_text(size = 6), #Legend text size
        legend.text=element_text(size = 6), #Legend text size
        legend.position = c(0.75, 0.9)) +
  ggtitle("i) CyclinB") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 9, 
                                  hjust = 0))
Mcap.CyclinB
```

Plot SMAUG expression as a function of time and treatment
```{r}
Mcap.SMG <- SMG.counts %>% subset(gene_id%in%top_SMG) %>% ggplot(aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(), lty="dashed", color = "darkgrey") +
  geom_point(aes(), size = 1.5) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (vst-transformed counts)"))) +
  theme_bw() + #Set the background color
  ylim(8.728599, 19.30752) +
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_blank(), #set x-axis title size
        axis.text.x=element_blank(), #set x-axis label size
        axis.text.y=element_text(size = 6), #set x-axis label size
        axis.title.y=element_blank(), #set x-axis label size
        axis.ticks.x=element_blank(), #No x-label ticks
        legend.text=element_text(size = 6), #Legend text size
        legend.title=element_text(size = 6), #Legend text size
        legend.position = c(0.85, 0.9)) +
  ggtitle("ii) SMAUG") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 9, 
                                  hjust = 0))
Mcap.SMG
```

Plot KAISO expression as a function of time and treatment
```{r}
Mcap.KAISO <- KAISO.counts %>% subset(gene_id%in%top_KAISO) %>% 
  ggplot(aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(), lty="dashed", color = "darkgrey") +
  geom_point(aes(), size = 1.5) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (vst-transformed counts)"))) +
  theme_bw() + #Set the background color
  ylim(8.728599, 19.30752) +
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_blank(), #set x-axis title size
        axis.text.x=element_blank(), #set x-axis label size
        axis.text.y=element_text(size = 6), #set x-axis label size
        axis.title.y=element_blank(), #set x-axis label size
        axis.ticks.x=element_blank(), #No x-label ticks
        legend.text=element_text(size = 6), #Legend text size
        legend.title=element_text(size = 6), #Legend text size
        legend.position = c(0.75, 0.9)) +
  ggtitle("iii) KAISO") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 9, 
                                  hjust = 0))
Mcap.KAISO
```

Plot SOX2 expression as a function of time and treatment
```{r}
Mcap.SOX2 <- SOX2.counts %>% subset(gene_id%in%top_SOX2) %>% 
  ggplot(aes(x=time_point, y=mean, group=gene_id)) +
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(lty="dashed", color = "darkgrey") +
  geom_point(aes(), size = 1.5) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (vst-transformed counts)"))) +
  theme_bw() + #Set the background color
  ylim(8.728599, 19.30752) +
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_text(size = 6), #set x-axis title size
        axis.text.x=element_text(angle = 90, vjust = 0.5, hjust=1, size = 6), #set x-axis label size
        axis.text.y=element_text(size = 6), #set x-axis label size
        axis.title.y=element_text(size = 6), #set y-axis title size
        axis.ticks.x=element_blank(), #No x-label ticks
        legend.text=element_text(size = 6), #Legend text size
        legend.title=element_text(size = 6), #Legend text size
        legend.position = c(0.75, 0.9)) +
  ggtitle("iv) SOX2") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 9, 
                                  hjust = 0))
Mcap.SOX2
```

Plot WNT8 expression as a function of time and treatment
```{r}
Mcap.WNT8 <- WNT8.counts %>% subset(gene_id%in%top_WNT8) %>% 
  ggplot(aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(lty="dashed", color = "darkgrey") +
  geom_point(aes(), size = 1.5) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (vst-transformed counts)"))) +
  theme_bw() + #Set the background color
  ylim(8.728599, 19.30752) +
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_text(size = 6), #set x-axis title size
        axis.text.x=element_text(angle = 90, vjust = 0.5, hjust=1, size = 6), #set x-axis label size
        axis.text.y=element_text(size = 6), #set x-axis label size
        axis.title.y=element_blank(), #set x-axis label size
        axis.ticks.x=element_blank(), #No x-label ticks
        legend.text=element_text(size = 6), #Legend text size
        legend.title=element_text(size = 6), #Legend text size
        legend.position = c(0.75, 0.9),
        legend.spacing.y = unit(0.01, 'cm')) +
  #guides(shape=guide_legend(ncol=2)) +
  ggtitle("v) WNT8") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 9, 
                                  hjust = 0))
Mcap.WNT8
```

Plot TBXT expression as a function of time and treatment
```{r}
Mcap.TBXT <- TBXT.counts %>% subset(gene_id%in%top_TBXT) %>% 
  ggplot(aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(lty="dashed", color = "darkgrey") +
  geom_point(aes(), size = 1.5) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (vst-transformed counts)"))) +
  theme_bw() + #Set the background color
  ylim(8.728599, 19.30752) +
    theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_text(size = 6), #set x-axis title size
        axis.text.x=element_text(angle = 90, vjust = 0.5, hjust=1, size = 6), #set x-axis label size
        axis.text.y=element_text(size = 6), #set x-axis label size
        axis.title.y=element_text(size = 6), #set x-axis label size
        axis.ticks.x=element_blank(), #No x-label ticks
        legend.text=element_text(size = 6), #Legend text size
        legend.title=element_blank(), #Legend text size
        legend.position = c(0.85, 0.85)) +
  ggtitle("vi) TBXT") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 9, 
                                  hjust = 0))
Mcap.TBXT
```

Compile plots into single figure and save
```{r}
Fig.6 <- cowplot::plot_grid(Mcap.CyclinB, Mcap.SMG, Mcap.KAISO, Mcap.SOX2, Mcap.WNT8, Mcap.TBXT, align = "vh", ncol = 3)
Fig.6
ggsave("2b-Epi-MZT-Enzyme-Expression/Output/Fig6-Mcap-Dev-MZT-Expression.pdf", Fig.6, width=7, height=5)
```

# Run differential expression analysis to see if any biomarkers are differentially expressed
Run differential expression test using a Wald model. 
```{r, message = FALSE}
DEG_dev <- DESeq(gdds_dev) #run differential expression test by group using the Wald model
```

Make a list of biomarkers to look for:
```{r}
Epi.markers <- c(top_DNMT1, top_DNMT3, TET1.counts$gene_id, MBD3.counts$gene_id, UHRF1.counts$gene_id, top_BRG1)

Dev.markers <- c(top_CyclinB, top_SMG, top_KAISO, top_SOX2, top_WNT8, top_TBXT)
```


Look for differential expression of biomarkers
```{r, warning=FALSE}
DESeq.results.fe_vs_ue <-  results(DEG_dev, name="time_point_Fertilized_egg_vs_Unfertilized_egg")
DESeq.results.fe_vs_ue <- as.data.frame(DESeq.results.fe_vs_ue)
DESeq.results.fe_vs_ue$contrast <- c("Fertilized_egg_vs_Unfertilized_egg")
DESeq.results.fe_vs_ue$gene_id <-  rownames(DESeq.results.fe_vs_ue)
Epi.DEGs.fe_vs_ue.1 <- filter(DESeq.results.fe_vs_ue, gene_id%in%Epi.markers, padj<0.05, log2FoldChange>1)
Epi.DEGs.fe_vs_ue.2 <- filter(DESeq.results.fe_vs_ue, gene_id%in%Epi.markers, padj<0.05, log2FoldChange<(-1))
Dev.DEGs.fe_vs_ue.1 <- filter(DESeq.results.fe_vs_ue, gene_id%in%Dev.markers, padj<0.05, log2FoldChange>1)
Dev.DEGs.fe_vs_ue.2 <- filter(DESeq.results.fe_vs_ue, gene_id%in%Dev.markers, padj<0.05, log2FoldChange<(-1))

DESeq.results.clvg_vs_fe <-  results(DEG_dev, contrast= c("time_point","Cleavage","Fertilized_egg"))
DESeq.results.clvg_vs_fe <- as.data.frame(DESeq.results.clvg_vs_fe)
DESeq.results.clvg_vs_fe$contrast <- c("Cleavage_vs_Fertilized_egg")
DESeq.results.clvg_vs_fe$gene_id <-  rownames(DESeq.results.clvg_vs_fe)
Epi.DEGs.clvg_vs_fe.1 <- filter(DESeq.results.clvg_vs_fe, gene_id%in%Epi.markers, padj<0.05, log2FoldChange>1)
Epi.DEGs.clvg_vs_fe.2 <- filter(DESeq.results.clvg_vs_fe, gene_id%in%Epi.markers, padj<0.05, log2FoldChange<(-1))
Dev.DEGs.clvg_vs_fe.1 <- filter(DESeq.results.clvg_vs_fe, gene_id%in%Dev.markers, padj<0.05, log2FoldChange>1)
Dev.DEGs.clvg_vs_fe.2 <- filter(DESeq.results.clvg_vs_fe, gene_id%in%Dev.markers, padj<0.05, log2FoldChange<(-1))

DESeq.results.pc_vs_clvg <-  results(DEG_dev, contrast= c("time_point","Prawn_chip","Cleavage"))
DESeq.results.pc_vs_clvg <- as.data.frame(DESeq.results.pc_vs_clvg)
DESeq.results.pc_vs_clvg$contrast <- c("Prawn_chip_vs_Cleavage")
DESeq.results.pc_vs_clvg$gene_id <-  rownames(DESeq.results.pc_vs_clvg)
Epi.DEGs.pc_vs_clvg.1 <- filter(DESeq.results.pc_vs_clvg, gene_id%in%Epi.markers, padj<0.05, log2FoldChange>1)
Epi.DEGs.pc_vs_clvg.2 <- filter(DESeq.results.pc_vs_clvg, gene_id%in%Epi.markers, padj<0.05, log2FoldChange<(-1))
Dev.DEGs.pc_vs_clvg.1 <- filter(DESeq.results.pc_vs_clvg, gene_id%in%Dev.markers, padj<0.05, log2FoldChange>1)
Dev.DEGs.pc_vs_clvg.2 <- filter(DESeq.results.pc_vs_clvg, gene_id%in%Dev.markers, padj<0.05, log2FoldChange<(-1))

DESeq.results.eg_vs_pc <-  results(DEG_dev, contrast= c("time_point", "Early_gastrula","Prawn_chip"))
DESeq.results.eg_vs_pc <- as.data.frame(DESeq.results.eg_vs_pc)
DESeq.results.eg_vs_pc$contrast <- c("Early_gastrula_vs_Prawn_chip")
DESeq.results.eg_vs_pc$gene_id <-  rownames(DESeq.results.eg_vs_pc)
Epi.DEGs.eg_vs_pc.1 <- filter(DESeq.results.eg_vs_pc, gene_id%in%Epi.markers, padj<0.05, log2FoldChange>1)
Epi.DEGs.eg_vs_pc.2 <- filter(DESeq.results.eg_vs_pc, gene_id%in%Epi.markers, padj<0.05, log2FoldChange<(-1))
Dev.DEGs.eg_vs_pc.1 <- filter(DESeq.results.eg_vs_pc, gene_id%in%Dev.markers, padj<0.05, log2FoldChange>1)
Dev.DEGs.eg_vs_pc.2 <- filter(DESeq.results.eg_vs_pc, gene_id%in%Dev.markers, padj<0.05, log2FoldChange<(-1))

DESeq.results.mg_vs_eg <-  results(DEG_dev, contrast= c("time_point", "Mid_gastrula", "Early_gastrula"))
DESeq.results.mg_vs_eg <- as.data.frame(DESeq.results.mg_vs_eg)
DESeq.results.mg_vs_eg$contrast <- c("Mid_gastrula_vs_Early_gastrula")
DESeq.results.mg_vs_eg$gene_id <-  rownames(DESeq.results.mg_vs_eg)
Epi.DEGs.mg_vs_eg.1 <- filter(DESeq.results.mg_vs_eg, gene_id%in%Epi.markers, padj<0.05, log2FoldChange>1)
Epi.DEGs.mg_vs_eg.2 <- filter(DESeq.results.mg_vs_eg, gene_id%in%Epi.markers, padj<0.05, log2FoldChange<(-1))
Dev.DEGs.mg_vs_eg.1 <- filter(DESeq.results.mg_vs_eg, gene_id%in%Dev.markers, padj<0.05, log2FoldChange>1)
Dev.DEGs.mg_vs_eg.2 <- filter(DESeq.results.mg_vs_eg, gene_id%in%Dev.markers, padj<0.05, log2FoldChange<(-1))

DESeq.results.lg_vs_mg <-  results(DEG_dev, contrast= c("time_point","Late_gastrula","Mid_gastrula"))
DESeq.results.lg_vs_mg <- as.data.frame(DESeq.results.lg_vs_mg)
DESeq.results.lg_vs_mg$contrast <- c("Late_gastrula_vs_Mid_gastrula")
DESeq.results.lg_vs_mg$gene_id <-  rownames(DESeq.results.lg_vs_mg)
Epi.DEGs.lg_vs_mg.1 <- filter(DESeq.results.lg_vs_mg, gene_id%in%Epi.markers, padj<0.05, log2FoldChange>1)
Epi.DEGs.lg_vs_mg.2 <- filter(DESeq.results.lg_vs_mg, gene_id%in%Epi.markers, padj<0.05, log2FoldChange<(-1))
Dev.DEGs.lg_vs_mg.1 <- filter(DESeq.results.lg_vs_mg, gene_id%in%Dev.markers, padj<0.05, log2FoldChange>1)
Dev.DEGs.lg_vs_mg.2 <- filter(DESeq.results.lg_vs_mg, gene_id%in%Dev.markers, padj<0.05, log2FoldChange<(-1))

DESeq.results.pln_vs_lg <-  results(DEG_dev, contrast= c("time_point", "Planula", "Late_gastrula"))
DESeq.results.pln_vs_lg <- as.data.frame(DESeq.results.pln_vs_lg)
DESeq.results.pln_vs_lg$contrast <- c("Planula_vs_Late_gastrula")
DESeq.results.pln_vs_lg$gene_id <-  rownames(DESeq.results.pln_vs_lg)
Epi.DEGs.pln_vs_lg.1 <- filter(DESeq.results.pln_vs_lg, gene_id%in%Epi.markers, padj<0.05, log2FoldChange>1)
Epi.DEGs.pln_vs_lg.2 <- filter(DESeq.results.pln_vs_lg, gene_id%in%Epi.markers, padj<0.05, log2FoldChange<(-1))
Dev.DEGs.pln_vs_lg.1 <- filter(DESeq.results.pln_vs_lg, gene_id%in%Dev.markers, padj<0.05, log2FoldChange>1)
Dev.DEGs.pln_vs_lg.2 <- filter(DESeq.results.pln_vs_lg, gene_id%in%Dev.markers, padj<0.05, log2FoldChange<(-1))

DESeq.results.adult_vs_pln <-  results(DEG_dev, contrast= c("time_point", "Adult","Planula"))
DESeq.results.adult_vs_pln <- as.data.frame(DESeq.results.adult_vs_pln)
DESeq.results.adult_vs_pln$contrast <- c("Adult_vs_Planula")
DESeq.results.adult_vs_pln$gene_id <-  rownames(DESeq.results.adult_vs_pln)
Epi.DEGs.adult_vs_pln.1 <- filter(DESeq.results.adult_vs_pln, gene_id%in%Epi.markers, padj<0.05, log2FoldChange>1)
Epi.DEGs.adult_vs_pln.2 <- filter(DESeq.results.adult_vs_pln, gene_id%in%Epi.markers, padj<0.05, log2FoldChange<(-1))
Dev.DEGs.adult_vs_pln.1 <- filter(DESeq.results.adult_vs_pln, gene_id%in%Dev.markers, padj<0.05, log2FoldChange>1)
Dev.DEGs.adult_vs_pln.2 <- filter(DESeq.results.adult_vs_pln, gene_id%in%Dev.markers, padj<0.05, log2FoldChange<(-1))
```

Bind results and add enzyme names
```{r}
Epi_DEGs <- bind_rows(Epi.DEGs.fe_vs_ue.1, Epi.DEGs.clvg_vs_fe.1, Epi.DEGs.pc_vs_clvg.1, Epi.DEGs.eg_vs_pc.1, Epi.DEGs.mg_vs_eg.1, Epi.DEGs.lg_vs_mg.1, Epi.DEGs.pln_vs_lg.1, Epi.DEGs.adult_vs_pln.1, Epi.DEGs.fe_vs_ue.2, Epi.DEGs.clvg_vs_fe.2, Epi.DEGs.pc_vs_clvg.2, Epi.DEGs.eg_vs_pc.2, Epi.DEGs.mg_vs_eg.2, Epi.DEGs.lg_vs_mg.2, Epi.DEGs.pln_vs_lg.2, Epi.DEGs.adult_vs_pln.2)
Epi_DEGs <- merge(Epi_DEGs, hits[,c(1,3)], by="gene_id")
head(Epi_DEGs)

Dev_DEGs <- bind_rows(Dev.DEGs.fe_vs_ue.1, Dev.DEGs.clvg_vs_fe.1, Dev.DEGs.pc_vs_clvg.1, Dev.DEGs.eg_vs_pc.1, Dev.DEGs.mg_vs_eg.1, Dev.DEGs.lg_vs_mg.1, Dev.DEGs.pln_vs_lg.1, Dev.DEGs.adult_vs_pln.1, Dev.DEGs.fe_vs_ue.2, Dev.DEGs.clvg_vs_fe.2, Dev.DEGs.pc_vs_clvg.2, Dev.DEGs.eg_vs_pc.2, Dev.DEGs.mg_vs_eg.2, Dev.DEGs.lg_vs_mg.2, Dev.DEGs.pln_vs_lg.2, Dev.DEGs.adult_vs_pln.2)
Dev_DEGs <- merge(Dev_DEGs, hits[,c(1,3)], by="gene_id")
head(Dev_DEGs)

write_csv(Epi_DEGs, file = "2b-Epi-MZT-Enzyme-Expression/Output/Epi_biomarker_DEGs.csv")
write_csv(Dev_DEGs, file = "2b-Epi-MZT-Enzyme-Expression/Output/Dev_biomarker_DEGs.csv")
```

