---
title: "MZT Biomarker Expression"
author: "Erin Chille"
date: "DRAFT: Last updated 2020/09/10"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Set up workspace

Remove prior objects and load necessary tools
```{r, warning=FALSE, message=FALSE}
rm(list=ls()) # removes all prior objects

library("RColorBrewer")
library("ggplot2")
library("tidyverse")
library("ggpubr")
```


Load sample Information
```{r}
sample.info <- read.csv(file="Data/RNAseq_data.csv", header=T, sep=",") #load sample info
```

Load gene count Information
```{r}
counts <- as.data.frame(read.csv("Data/gene_abund_TPM.csv", row.names="gene_id"), colClasses = double) #Load normalized expression matrix from stringtie
```


Keep only ambient treatmentinfo and count data
```{r}
dev <- c("AMB")
treatmentinfo_dev <- filter(sample.info, treatment %in% dev)
dim(treatmentinfo_dev) #rows should be 24

gcount_dev <- counts[treatmentinfo_dev$sample_id]
dim(gcount_dev) #columns should be 24
```

## Run BLAST search of biomarkers against Mcap protein fasta

Find orthologues of biomarkers within Mcap genome
```{bash}
##Make blast database from Mcap predicted protein fasta file
#/usr/local/ncbi/blast/bin/makeblastdb -in Data/Mcap.protein.fa -input_type fasta -dbtype prot -title Mcap.protein -parse_seqids -out /Data/Mcap.protein.blast.db
#
##Blast fasta file containing sequences of selected biomarkers against our sequences
#/usr/local/ncbi/blast/bin/blastp -query Data/MZT_biomarkers.fa -db Data/Mcap.protein.blast.db -outfmt 6 -max_target_seqs 5 -evalue 1e-05 -out Output/RNAseq/Mcap_genome_MZT_biomarker_hits_10
```

Subset results for genes of interest
```{r}
hits <- read.table(file="Output/RNAseq/Mcap_genome_MZT_biomarker_hits_10.tsv", header=F, sep="\t")
colnames(hits) <- c("Epi.gene","queryprot", "gene_id", "pident", "length", "mismatch", "gapopen", "qstart", "qend", "sstart", "send", "evalue", "bitscore")
hits <- hits %>% distinct(Epi.gene, gene_id, .keep_all = TRUE)

countsx <- gcount_dev
gene.counts <- countsx[(row.names(countsx) %in% hits$gene), ]
str(gene.counts)
gene.counts$gene_id <- row.names(gene.counts) 
gene.counts <- merge(gene.counts, hits[,c(1,3)], by="gene_id")

gene.counts.long <- pivot_longer(gene.counts, 2:25, values_to = "exp")
  #gather(gene.counts, group, exp, X1101:X375, factor_key=TRUE) 
colnames(gene.counts.long)[3] <- "sample_id"
gene.counts.long <- merge(gene.counts.long, sample.info[,c(1:2)], by="sample_id")
str(gene.counts.long)

mean.counts <- gene.counts.long %>%
  group_by(gene_id, Epi.gene, time_point) %>%
  summarise(mean = mean(exp),
            sd=sd(exp))

mean.counts$time_point <- factor(mean.counts$time_point, 
                                 levels = c("Unfertilized_egg","Fertilized_egg","Cleavage", 
                                  "Prawn_chip","Early_gastrula","Mid_gastrula", "Late_gastrula",
                                  "Planula", "Adult"))

DNMT3.counts <- subset(mean.counts, Epi.gene =="DNMT3A")
DNMT1.counts <- subset(mean.counts, Epi.gene =="DNMT1")
TET1.counts <- subset(mean.counts, Epi.gene =="TET1")
MBD2.counts <- subset(mean.counts, Epi.gene =="MBD2")
MBD3.counts <- subset(mean.counts, Epi.gene =="MBD3")
UHRF1.counts <- subset(mean.counts, Epi.gene =="UHRF1")
BRG1.counts <- subset(mean.counts, Epi.gene =="BRG1")
SMG.counts <- subset(mean.counts, Epi.gene =="SMG") #SMAUG
KAISO.counts <- subset(mean.counts, Epi.gene =="KAISO") #Kaiso
NANOG.counts <- subset(mean.counts, Epi.gene =="NANOG") #SNANOG
SOX2.counts <- subset(mean.counts, Epi.gene =="SOX2") #SOX2
WNT8.counts <- subset(mean.counts, Epi.gene =="WNT8") #WNT8
TBXT.counts <- subset(mean.counts, Epi.gene =="TBXT") #brachyury
```

## Plot methylation-related genes

Plot DNMT3a expression as a function of time and treatment
```{r}
# pdf("Output/Mcap_Dev_DNMT3A_expression.pdf")
Mcap.DNMT3 <- ggplot(DNMT3.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), colour="black", width=.1) +
  geom_line(aes(), lty="dashed") +
  geom_point(aes(shape=gene_id), size = 2) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.title.x=element_blank(), #No x-title
        axis.text.x=element_blank(), #No x-labels
        axis.ticks.x=element_blank(), #No x-label ticks
        axis.title.y=element_text(size = 12), #size = 8), #set y-axis title size
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank()) + #Set the plot background
        #legend.position="none") + #set legend
  ggtitle("i) DNMT3a  homologs") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  #theme(strip.text.y = element_text(size = 8)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 16, 
                                  hjust = 0))
Mcap.DNMT3
dev.off()

bMcap.DNMT3 <- ggplot(DNMT3.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(), lty="dashed") +
  geom_point(aes(shape=gene_id), size = 2) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_blank(), #set x-axis title size
        axis.text.x=element_blank(), #set x-axis label size
        axis.ticks.x=element_blank()) + #No x-label ticks
  ggtitle("i) DNMT3A homologs") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
bMcap.DNMT3
```

Plot DNMT1 expression as a function of time and treatment
```{r}
# pdf("Output/Mcap_Dev_DNMT1_expression.pdf")
Mcap.DNMT1 <- ggplot(DNMT1.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), colour="black", width=.1) +
  geom_line(aes(), lty="dashed") +
  geom_point(aes(), size = 2, ) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="none") + #set legend
  ggtitle("ii) DNMT1 homologs") +
  facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
Mcap.DNMT1
#dev.off()

bMcap.DNMT1 <- ggplot(DNMT1.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(), lty="dashed") +
  geom_point(aes(shape=gene_id), size = 2) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_blank(), #set x-axis title size
        axis.title.y=element_blank(), #set y-axis title size
        axis.text.x=element_blank(), #set x-axis label size
        axis.ticks.x=element_blank()) + #No x-label ticks
  ggtitle("ii) DNMT1 homologs") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
bMcap.DNMT1
```

Plot TET1 expression as a function of time and treatment
```{r}
# pdf("Output/Mcap_Dev_TET1_expression.pdf")
Mcap.TET1 <- ggplot(TET1.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), colour="black", width=.1) +
  geom_point(aes(), size = 2 ) +
  geom_line(aes(), lty="dashed") +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="none") + #set legend
  ggtitle("iii) TET1 homologs") +
  facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
Mcap.TET1
dev.off()

bMcap.TET1 <- ggplot(TET1.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(), lty="dashed") +
  geom_point(aes(shape=gene_id), size = 2) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_blank(), #set x-axis title size
        axis.title.y=element_blank(), #set y-axis title size
        axis.text.x=element_blank(), #set x-axis label size
        axis.ticks.x=element_blank()) + #No x-label ticks
  ggtitle("iii) TET1 homologs") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
bMcap.TET1
```

Plot MBD2 expression as a function of time and treatment
```{r}
# pdf("Output/Mcap_Dev_MBD2_expression.pdf")
Mcap.MBD2 <- ggplot(MBD2.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), colour="black", width=.1) +
  geom_point(aes(), size = 2 ) +
  geom_line(aes(), lty="dashed") +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="none") + #set legend
  ggtitle("iv) MBD2/3 homologs") +
  facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
Mcap.MBD2
dev.off()

bMcap.MBD2 <- ggplot(MBD2.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(shape=gene_id), lty="dashed") +
  geom_point(aes(shape=gene_id), size = 2, ) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=9), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="right") + #set legend
  ggtitle("iv) MBD2/3 homologs") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
bMcap.MBD2
```

Plot MBD3 expression as a function of time and treatment
```{r}
# pdf("Output/Mcap_Dev_MBD3_expression.pdf")
# Mcap.MBD3 <- ggplot(MBD3.counts, aes(x=time_point, y=mean, group=gene_id)) + 
#   geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), colour="black", width=.1) +
#   geom_point(aes(), size = 2 ) +
#   geom_line(aes(), lty="dashed") +
#   xlab("Timepoint") +
#   ylab(expression(paste("Mean expression (TPM)"))) +
#   theme_bw() + #Set the background color
#   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
#         axis.line = element_line(color = 'black'), #Set the axes color
#         panel.border = element_blank(), #Set the border
#         panel.grid.major = element_blank(), #Set the major gridlines
#         panel.grid.minor = element_blank(), #Set the minor gridlines
#         plot.background=element_blank(),  #Set the plot background
#         legend.position="none") + #set legend
#   ggtitle("A) M. capitata MBD3 homologs") +
#   facet_grid(rows=vars(gene_id), scales="free_y") +
#   theme(strip.text.y = element_text(size = 5)) +
#   theme(plot.title = element_text(face = 'bold.italic', 
#                                   size = 14, 
#                                   hjust = 0))
# Mcap.MBD3
# dev.off()
# 
# ggplot(MBD3.counts, aes(x=time_point, y=mean, group=gene_id)) + 
#   geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), colour="black", width=.1) +
#   geom_point(aes(), size = 2 ) +
#   geom_line(aes(), lty="dashed") +
#   xlab("Timepoint") +
#   ylab(expression(paste("Mean expression (TPM)"))) +
#   theme_bw() + #Set the background color
#   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
#         axis.line = element_line(color = 'black'), #Set the axes color
#         panel.border = element_blank(), #Set the border
#         panel.grid.major = element_blank(), #Set the major gridlines
#         panel.grid.minor = element_blank(), #Set the minor gridlines
#         plot.background=element_blank(),  #Set the plot background
#         legend.position="none") + #set legend
#   ggtitle("A) M. capitata MBD3 homologs") +
#   facet_grid(rows=vars(gene_id), scales="free_y") +
#   theme(strip.text.y = element_text(size = 5)) +
#   theme(plot.title = element_text(face = 'bold.italic', 
#                                   size = 14, 
#                                   hjust = 0))
```

Plot UHRF1 expression as a function of time and treatment
```{r}
# pdf("Output/Mcap_Dev_UHRF1_expression.pdf")
Mcap.UHRF1 <- ggplot(UHRF1.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), colour="black", width=.1) +
  geom_point(aes(), size = 2 ) +
  geom_line(aes(), lty="dashed") +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="none") + #set legend
  ggtitle("v) UHRF1 homologs") +
  facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
Mcap.UHRF1
dev.off()

bMcap.UHRF1 <- ggplot(UHRF1.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(shape=gene_id), lty="dashed") +
  geom_point(aes(shape=gene_id), size = 2, ) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=9), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        axis.title.y=element_blank(), #set y-axis title size
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="right") + #set legend
  ggtitle("v) UHRF1 homologs") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
bMcap.UHRF1
```

Plot BRG1 expression as a function of time and treatment
```{r}
# pdf("Output/Mcap_Dev_BRG1_expression.pdf")
Mcap.BRG1 <- ggplot(BRG1.counts, aes(x=time_point, y=mean, group=gene_id)) +
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), colour="black", width=.1) +
  geom_point(aes(), size = 2 ) +
  geom_line(aes(), lty="dashed") +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="none") + #set legend
  ggtitle("vi) BRG1 homologs") +
  facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic',
                                  size = 14,
                                  hjust = 0))
Mcap.BRG1
dev.off()

bMcap.BRG1 <- ggplot(BRG1.counts, aes(x=time_point, y=mean, group=gene_id)) +
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(shape=gene_id), lty="dashed") +
  geom_point(aes(shape=gene_id), size = 2, ) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=9), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        axis.title.y=element_blank(), #set y-axis title size
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="right") + #set legend
  ggtitle("vi) BRG1 homologs") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic',
                                  size = 14,
                                  hjust = 0))
bMcap.BRG1
```

Compile plots into single figure and save
```{r}
# Fig.5A <- cowplot::plot_grid(Mcap.DNMT3, Mcap.DNMT1, Mcap.TET1, Mcap.MBD2, Mcap.UHRF1, Mcap.BRG1, align = "v", ncol = 3)
# Fig.5A
# ggsave("Output/Mcap.Dev.Epi.Expression.pdf", Fig.5A, width=21, height=14)

bFig.5A <- cowplot::plot_grid(bMcap.DNMT3, bMcap.DNMT1, bMcap.TET1, bMcap.MBD2, bMcap.UHRF1, bMcap.BRG1, align = "v", ncol = 3)
bFig.5A
ggsave("Output/bMcap.Dev.Epi.Expression.pdf", bFig.5A, width=21, height=15)
```

## Plot MZT biomarkers

Plot SMAUG expression as a function of time and treatment
```{r}
# pdf("Output/Mcap_Dev_SMG_expression.pdf")
Mcap.SMG <- ggplot(SMG.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), colour="black", width=.1) +
  geom_point(aes(), size = 2 ) +
  geom_line(aes(), lty="dashed") +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="none") + #set legend
  ggtitle("i) SMAUG homologs") +
  facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
Mcap.SMG
dev.off()

bMcap.SMG <- ggplot(SMG.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(), lty="dashed") +
  geom_point(aes(shape=gene_id), size = 2) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.x=element_blank(), #set x-axis title size
        axis.text.x=element_blank(), #set x-axis label size
        axis.ticks.x=element_blank()) + #No x-label ticks
  ggtitle("i) SMAUG homologs") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
bMcap.SMG
```

Plot KAISO expression as a function of time and treatment
```{r}
# pdf("Output/Mcap_Dev_KAISO_expression.pdf")
Mcap.KAISO <- ggplot(KAISO.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), colour="black", width=.1) +
  geom_point(aes(), size = 2 ) +
  geom_line(aes(), lty="dashed") +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="none") + #set legend
  ggtitle("ii) KAISO homologs") +
  facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
Mcap.KAISO
dev.off()

bMcap.KAISO <- ggplot(KAISO.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(), lty="dashed") +
  geom_point(aes(shape=gene_id), size = 2) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        axis.title.y=element_blank(), #set y-axis title size
        axis.title.x=element_blank(), #set x-axis title size
        axis.text.x=element_blank(), #set x-axis label size
        axis.ticks.x=element_blank()) + #No x-label ticks
  ggtitle("ii) KAISO homologs") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
bMcap.KAISO
```

Plot TBXT expression as a function of time and treatment
```{r}
# pdf("Output/Mcap_Dev_TBXT_expression.pdf")
Mcap.TBXT <- ggplot(TBXT.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), colour="black", width=.1) +
  geom_point(aes(), size = 2 ) +
  geom_line(aes(), lty="dashed") +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="none") + #set legend
  ggtitle("iii) Brachyury (TBXT) homologs") +
  facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
Mcap.TBXT
dev.off()

bMcap.TBXT <- ggplot(TBXT.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(shape=gene_id), lty="dashed") +
  geom_point(aes(shape=gene_id), size = 2, ) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=9), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        axis.title.y=element_blank(), #set y-axis title size
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="right") + #set legend
  ggtitle("iii) Brachyury (TBXT) homologs") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
bMcap.TBXT
```

Plot NANOG expression as a function of time and treatment
```{r}
# pdf("Output/Mcap_Dev_NANOG_expression.pdf")
Mcap.NANOG <- ggplot(NANOG.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), colour="black", width=.1) +
  geom_point(aes(), size = 2 ) +
  geom_line(aes(), lty="dashed") +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="none") + #set legend
  ggtitle("iv) NANOG homologs") +
  facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
Mcap.NANOG
dev.off()

bMcap.NANOG <- ggplot(NANOG.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(shape=gene_id), lty="dashed") +
  geom_point(aes(shape=gene_id), size = 2, ) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=9), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        axis.title.y=element_blank(), #set y-axis title size
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="right") + #set legend
  ggtitle("iv) NANOG homologs") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
bMcap.NANOG
```


Plot SOX2 expression as a function of time and treatment
```{r}
# pdf("Output/Mcap_Dev_SOX2_expression.pdf")
Mcap.SOX2 <- ggplot(SOX2.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), colour="black", width=.1) +
  geom_point(aes(), size = 2 ) +
  geom_line(aes(), lty="dashed") +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="none") + #set legend
  ggtitle("v) SOX2 homologs") +
  facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
Mcap.SOX2
dev.off()

bMcap.SOX2 <- ggplot(SOX2.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(shape=gene_id), lty="dashed") +
  geom_point(aes(shape=gene_id), size = 2, ) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=9), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="right") + #set legend
  ggtitle("v) SOX2 homologs") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
bMcap.SOX2
```


Plot WNT8 expression as a function of time and treatment
```{r}
# pdf("Output/Mcap_Dev_WNT8_expression.pdf")
Mcap.WNT8 <- ggplot(WNT8.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), colour="black", width=.1) +
  geom_point(aes(), size = 2 ) +
  geom_line(aes(), lty="dashed") +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="none") + #set legend
  ggtitle("vi) WNT8 homologs") +
  facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
Mcap.WNT8
dev.off()

bMcap.WNT8 <- ggplot(WNT8.counts, aes(x=time_point, y=mean, group=gene_id)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.1) +
  geom_line(aes(shape=gene_id), lty="dashed") +
  geom_point(aes(shape=gene_id), size = 2, ) +
  xlab("Timepoint") +
  ylab(expression(paste("Mean expression (TPM)"))) +
  theme_bw() + #Set the background color
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=9), #Set the text angle
        axis.line = element_line(color = 'black'), #Set the axes color
        panel.border = element_blank(), #Set the border
        axis.title.y=element_blank(), #set y-axis title size
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        plot.background=element_blank(),  #Set the plot background
        legend.position="right") + #set legend
  ggtitle("vi) WNT8 homologs") +
  #facet_grid(rows=vars(gene_id), scales="free_y") +
  theme(strip.text.y = element_text(size = 5)) +
  theme(plot.title = element_text(face = 'bold.italic', 
                                  size = 14, 
                                  hjust = 0))
bMcap.WNT8
```

Compile plots into single figure and save
```{r}
# Fig.5B <- cowplot::plot_grid(Mcap.SMG, Mcap.KAISO, Mcap.TBXT, Mcap.SOX2, Mcap.NANOG, Mcap.WNT8, align = "v", ncol = 3)
# Fig.5B
# ggsave("Output/Mcap.Dev.MZT.Expression.pdf", Fig.5B, width=21, height=14)
bFig.5B <- cowplot::plot_grid(bMcap.SMG, bMcap.KAISO, bMcap.TBXT, bMcap.SOX2, bMcap.NANOG, bMcap.WNT8, align = "v", ncol = 3)
bFig.5B
ggsave("Output/bMcap.Dev.MZT.Expression.pdf", bFig.5B, width=21, height=14)
```

